prov/mrail/src/mrail_init.c

```diff
@@ -85,6 +85,12 @@ int mrail_get_core_info(uint32_t version, const char *node, const char *service,
 	if (!core_hints)
 		return -FI_ENOMEM;
 
+	if (!hints) {
+		core_hints->mode = ~0ULL;
```

Comment:
We should be explicit in the modes that either mrail supports or what it can pass through to the caller.  This isn't future proof, since we have no idea what new mode bits might mean.

Reply:
This is a special case of NULL hints so we set all modes. In case the app passes a valid hints, then we would set only the modes we support.

Reply:
But that could return an fi_info that mrail can't actually support.

Reply:
Yes but that would exactly mirror the app's request. The app passed a NULL hints which implies that the app could get modes it may not be able to support.